google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/googleapis/DefaultGoogleApiFactory.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.googleapis;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import java.time.Instant;
+import java.util.Date;
+
+/** Default implementation for instantiating various Google APIs. */
+public class DefaultGoogleApiFactory extends GoogleApiFactory {
+
+  @Override
+  public Storage newStorageApi(String projectId, Credential credential) {
+    return StorageOptions.newBuilder()
+        .setProjectId(projectId)
+        .setCredentials(
+            new GoogleCredentials(
+                new AccessToken(
+                    credential.getAccessToken(),
+                    credential.getExpirationTimeMilliseconds() != null
```

Comment:
Can you break out this clause (the expiration time) into a local variable for better readability?